### DISCUSSION TITLE
Kibana returning zero logs in fresh 2.3.30 distributed installation.

### DISCUSSION THREAD

**Discussion by InfosecGoon** [Security-Onion-Solutions/securityonion#3306] [Unsupported Versions] [answered]
I'm replacing our current distributed installation of Security Onion 16.04 with one of 2.3.30, and I've run into a bit of an issue with Kibana. I would appreciate any light that anyone could shed on it.

The current 2.3.30 installation is a Manager node running in VMWare, and two physical nodes -- one a Forward node hooked to our tap infrastructure, and one a Search node in the same datacenter as the Forward node. The Manager node was originally built using the preceding version of SO 2 (2.3.21, I believe?), then updated using soup before the physical nodes were added. The physical nodes were installed from the 2.3.30 ISO.

All of the nodes show up properly in the Grid view in SOC, and checking the Search node in Grafana shows that there are millions of records in the Elasticsearch indices. So the data from the Forward node is definitely being written into the Elasticsearch tables on the Search node. But all of the Kibana dashboards show no logs, and when I search I get entries like this in the Kibana log file in /opt/so/log/kibana:

`{"type":"log","@timestamp":"2021-03-04T21:49:42Z","tags":["error","elasticsearch","data"],"pid":292,"message":"[version_conflict_engine_exception]: [search-telemetry:search-telemetry]: version conflict, required seqNo [2374], primary term [4]. current document has seqNo [2378] and primary term [4]"}`

It appears to me that the Kibana queries are failing due to this version conflict exception, and so the interface is showing 0 logs. Something must be out of sync - does anyone know how I might fix this?

Thanks for your time,

--Matt

**Comment by bicfordio** [comment]
No clue how to fix, but encountering the exact same issue. I tried the installation of 2.3.21, was able to get the remote nodes to sync with the manager node, and was able to orchestrate the upgrade to 2.3.30 from the manager node. My remote nodes have millions of alerts logged but the manager node is unable to display any of them within the dashboard.

I am going to attempt a fresh install at 2.3.30 to see if there is any difference, but seems like that is what you have done too.

**Reply by PacketEng** [reply]
Same thing on my end as well, new install distributed tcpdump see activity nothing indexed or shipped from what I can tell.  Second full reinstall still same results.  This is a bummer :(

**Comment by bicfordio** [comment]
I initiated a fresh install of the manager node and a fresh install of the heavy node on 2.3.30, and I am able to see the logs being returned via Kibana and the manager node displaying alerts. However, I am encountering the problem seen within this thread on the Kibana web page not displaying:
https://github.com/Security-Onion-Solutions/securityonion/discussions/3280

**Comment by InfosecGoon** [comment, marked as answer]
An update on this: I had our VMWare team build a fresh manager node on 2.3.30 and rebuilt both the Forward and Search nodes from scratch. I'm still getting the same symptom -- Grafana shows that there are records in the Search node Elasticsearch installation, but Kibana on the manager node cannot retrieve them. The same "version conflict" records came up in the Kibana logs.

I tried to refresh the so-* index in Kibana, and it wiped out all of the attributes; could the problem be that it's just not able to contact Elasticsearch at all?

Any help would be appreciated. Thanks!

**Reply by TOoSmOotH** [reply]
What does the elasticsearch log say?

**Reply by InfosecGoon** [reply]
The Elasticsearch log on the manager node seems to indicate that the connection is timing out:

`[2021-03-12T20:27:02,306][WARN ][org.elasticsearch.transport.SniffConnectionStrategy] fetching nodes from external cluster [NODENAME_searchnode] failed
org.elasticsearch.transport.ConnectTransportException: [][10.200.0.166:9300] connect_timeout[30s]
        at org.elasticsearch.transport.TcpTransport$ChannelsConnectedListener.onTimeout(TcpTransport.java:984) ~[elasticsearch-7.10.2.jar:7.10.2]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:684) ~[elasticsearch-7.10.2.jar:7.10.2]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
        at java.lang.Thread.run(Thread.java:832) [?:?]
`

I'm able to curl https://localhost:9300 from a session on the search node, but I can't do it from the manager or from the Forward node that is on the same switch with no intervening firewall.

**Reply by InfosecGoon** [reply]
Additional update: Thanks to @TOoSmOotH for pointing me to the Elasticsearch logs, I was able to work with our firewall team to find an internal device that was blocking port 9300 traffic between the manager node and the search node. I think we're good for now and the issue is solved. Thanks!
